### ISSUE TITLE
🐛 Using qualityPrioritization = 'quality' will crash the app

### ISSUE THREAD

**Issue by andreiwow2** [cuvent/react-native-vision-camera#1415]
### What were you trying to do?

I tried to take a picture while using qualityPriuoritization set to quality.

### Reproduceable Code

```tsx
const takePhoto = async () => {
    try {
      if (cameraRef.current) {
        const centerPoint = {
          x: Dimensions.get('window').width / 2,
          y: Dimensions.get('window').height / 2,
        };
        await cameraRef.current.focus(centerPoint);
        const photo = await cameraRef.current.takePhoto({
          flash: useFlash ? 'on' : 'off',
          qualityPrioritization: 'quality',
        });
        toast.close(takingPhotoToastRef.current);
        setTakenPhoto(photo);
        takingPhotoToastRef.current = toast.show({
          title: 'Foto aufgenommen',
          placement: 'top',
          duration: 3000,
        });
        setCurrentlyTakingPhoto(false);
      }
    } catch (err) {
      console.log(err);
      takingPhotoToastRef.current = toast.show({
        title: 'Foto aufgenommen',
        placement: 'top',
        description: `${err}`,
        duration: 3000,
      });
    }
  };
```

### What happened instead?

App crashed with error in xCode

### Relevant log output

```shell
flipper: FlipperClient::addPlugin Inspector
flipper: FlipperClient::addPlugin Preferences
flipper: FlipperClient::addPlugin React
flipper: FlipperClient::addPlugin Network
2023-01-09 17:02:48.218518+0100 KareOcrBusinessCards[24531:3984908] [] nw_endpoint_copy_association_with_evaluator nw_context_caches_are_shared(context (<nw_context com.apple.CFNetwork.NSURLSession.{C76AB4EF-317C-4DDF-A320-B29DF9620826}{(null)}{Y}{2} (private)>), endpoint->context (<nw_context Default Network Context (private)>)) is false
2023-01-09 17:02:48.219570+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_socket_handle_socket_event [C1.1.1:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:48.220978+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_socket_handle_socket_event [C1.1.2:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:48.241201+0100 KareOcrBusinessCards[24531:3984907] [connection] nw_connection_get_connected_socket_block_invoke [C1] Client called nw_connection_get_connected_socket on unconnected nw_connection
2023-01-09 17:02:48.241233+0100 KareOcrBusinessCards[24531:3984907] TCP Conn 0x28109c6e0 Failed : error 0:61 [61]
2023-01-09 17:02:48.270815+0100 KareOcrBusinessCards[24531:3984908] Invalid encryption level
2023-01-09 17:02:48.288751+0100 KareOcrBusinessCards[24531:3984908] Invalid encryption level
2023-01-09 17:02:48.288823+0100 KareOcrBusinessCards[24531:3984908] Invalid encryption level
2023-01-09 17:02:48.303830+0100 KareOcrBusinessCards[24531:3984908] [] nw_protocol_default_input_finished called with null protocol->default_input_handler
2023-01-09 17:02:48.307947+0100 KareOcrBusinessCards[24531:3984908] [] nw_protocol_default_input_finished called with null protocol->default_input_handler, dumping backtrace:
        [arm64] libnetcore-3013.42.2
    0   Network                             0x00000001b906ed44 __nw_create_backtrace_string + 192
    1   Network                             0x00000001b8cbc630 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2762288
    2   Network                             0x00000001b927ca6c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8792684
    3   Network                             0x00000001b8e57e5c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 4447836
    4   Network                             0x00000001b923f1b0 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8540592
    5   Network                             0x00000001b923e908 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8538376
    6   Network                             0x00000001b910a444 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 7275588
    7   Network                             0x00000001b924b1a8 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8589736
    8   Network                             0x00000001b9249d2c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8584492
    9   Network                             0x00000001b9246558 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8570200
    10  Network                             0x00000001b9264358 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8692568
    11  Network                             0x00000001b8c5fc88 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2382984
    12  libdispatch.dylib                   0x000000010874c5a8 _dispatch_call_block_and_release + 32
    13  libdispatch.dylib                   0x000000010874e05c _dispatch_client_callout + 20
    14  libdispatch.dylib                   0x0000000108758700 _dispatch_workloop_invoke + 2952
    15  libdispatch.dylib                   0x0000000108763cbc _dispatch_workloop_worker_thread + 740
    16  libsystem_pthread.dylib             0x0000000205697df8 _pthread_wqthread + 288
    17  libsystem_pthread.dylib             0x0000000205697b98 start_wqthread + 8
2023-01-09 17:02:48.308225+0100 KareOcrBusinessCards[24531:3984908] [privacy_proxy] [C2.1.1:2] Received response code 503 for 10.10.100.49:8081, unreachable through proxy
2023-01-09 17:02:48.328654+0100 KareOcrBusinessCards[24531:3984695] [native] Running application KareOcrBusinessCards ({
    initialProps =     {
    };
    rootTag = 1;
})
2023-01-09 17:02:48.357220+0100 KareOcrBusinessCards[24531:3984905] Reanimated not found, Frame Processors are disabled.
2023-01-09 17:02:48.357729+0100 KareOcrBusinessCards[24531:3984910] [privacy_proxy] [C5.1.1:2] Received response code 503 for 10.10.100.49:8081, unreachable through proxy
2023-01-09 17:02:48.358276+0100 KareOcrBusinessCards[24531:3984910] [] __nwlog_err_simulate_crash simulate crash already simulated "nw_protocol_default_input_finished called with null protocol->default_input_handler"
2023-01-09 17:02:48.359709+0100 KareOcrBusinessCards[24531:3984910] [] nw_protocol_default_input_finished called with null protocol->default_input_handler, dumping backtrace:
        [arm64] libnetcore-3013.42.2
    0   Network                             0x00000001b906ed44 __nw_create_backtrace_string + 192
    1   Network                             0x00000001b8cbc630 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2762288
    2   Network                             0x00000001b9282b40 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8817472
    3   Network                             0x00000001b8c2bb6c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2169708
    4   Network                             0x00000001b9298a10 nw_hash_table_apply + 164
    5   Network                             0x00000001b8c2ab10 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2165520
    6   Network                             0x00000001b8c69f0c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2424588
    7   Network                             0x00000001b8c7333c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2462524
    8   Network                             0x00000001b924d5cc 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8598988
    9   Network                             0x00000001b927c7d0 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8792016
    10  Network                             0x00000001b8e57e5c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 4447836
    11  Network                             0x00000001b923f1b0 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 8540592
    12  Network                             0x00000001b8c2b6e0 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2168544
    13  Network                             0x00000001b9298a10 nw_hash_table_apply + 164
    14  Network                             0x00000001b8c2ab10 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2165520
    15  Network                             0x00000001b8c69f0c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2424588
    16  Network                             0x00000001b8c69534 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2422068
    17  Network                             0x00000001b8e1ec64 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 4213860
    18  Network                             0x00000001b9299054 nw_hash_table_apply + 1768
    19  Network                             0x00000001b8e1dd9c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 4210076
    20  Network                             0x00000001b8c6abec 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2427884
    21  Network                             0x00000001b8c69f00 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2424576
    22  Network                             0x00000001b8c69534 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 2422068
    23  Network                             0x00000001b954987c 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 11729020
    24  Network                             0x00000001b95473a8 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 11719592
    25  Network                             0x00000001b95465b4 3BF445F9-D58F-3280-B9C9-2FEAF4DACA6D + 11716020
    26  libdispatch.dylib                   0x000000010874e05c _dispatch_client_callout + 20
    27  libdispatch.dylib                   0x00000001087511e8 _dispatch_continuation_pop + 792
    28  libdispatch.dylib                   0x0000000108767ed0 _dispatch_source_invoke + 1668
    29  libdispatch.dylib                   0x00000001087584b8 _dispatch_workloop_invoke + 2368
    30  libdispatch.dylib                   0x0000000108763cbc _dispatch_workloop_worker_thread + 740
    31  libsystem_pthread.dylib             0x0000000205697df8 _pthread_wqthread + 288
    32  libsystem_pthread.dylib             0x0000000205697b98 start_wqthread + 8
Thread Performance Checker: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions
PID: 24531, TID: 3984695
Backtrace
=================================================================
3   KareOcrBusinessCards                0x00000001055a2910 -[_RCTSRRunLoopThread runLoop] + 44
4   KareOcrBusinessCards                0x00000001055a2640 __49+[NSRunLoop(RCTSRWebSocket) RCTSR_networkRunLoop]_block_invoke + 116
5   libdispatch.dylib                   0x000000010874e05c _dispatch_client_callout + 20
6   libdispatch.dylib                   0x000000010874fbcc _dispatch_once_callout + 136
7   KareOcrBusinessCards                0x00000001055a25a4 +[NSRunLoop(RCTSRWebSocket) RCTSR_networkRunLoop] + 84
8   KareOcrBusinessCards                0x000000010559c5e0 -[RCTSRWebSocket _connect] + 68
9   KareOcrBusinessCards                0x000000010559b384 -[RCTSRWebSocket open] + 300
10  KareOcrBusinessCards                0x000000010557de44 -[RCTReconnectingWebSocket start] + 148
11  KareOcrBusinessCards                0x000000010556d7f0 -[RCTPackagerConnection init] + 404
12  KareOcrBusinessCards                0x000000010556d63c __49+[RCTPackagerConnection sharedPackagerConnection]_block_invoke + 36
13  libdispatch.dylib                   0x000000010874e05c _dispatch_client_callout + 20
14  libdispatch.dylib                   0x000000010874fbcc _dispatch_once_callout + 136
15  KareOcrBusinessCards                0x000000010556d5f0 +[RCTPackagerConnection sharedPackagerConnection] + 84
16  KareOcrBusinessCards                0x0000000105607e10 -[RCTDevSettings initialize] + 148
17  KareOcrBusinessCards                0x00000001055569d4 -[RCTModuleData _initializeModule] + 96
18  KareOcrBusinessCards                0x0000000105556334 -[RCTModuleData setUpInstanceAndBridge:] + 2120
19  KareOcrBusinessCards                0x0000000105557dec -[RCTModuleData instance] + 1168
20  KareOcrBusinessCards                0x00000001054f29e8 -[RCTCxxBridge moduleForName:lazilyLoadIfNecessary:] + 680
21  KareOcrBusinessCards                0x0000000105560cb4 -[RCTModuleRegistry moduleForName:lazilyLoadIfNecessary:] + 128
22  KareOcrBusinessCards                0x0000000105560c28 -[RCTModuleRegistry moduleForName:] + 48
23  KareOcrBusinessCards                0x000000010561c784 -[RCTPerfMonitor devMenuItem] + 84
24  KareOcrBusinessCards                0x000000010561c658 -[RCTPerfMonitor initialize] + 88
25  KareOcrBusinessCards                0x00000001055569d4 -[RCTModuleData _initializeModule] + 96
26  KareOcrBusinessCards                0x0000000105556334 -[RCTModuleData setUpInstanceAndBridge:] + 2120
27  KareOcrBusinessCards                0x000000010555801c __25-[RCTModuleData instance]_block_invoke + 44
28  KareOcrBusinessCards                0x00000001055c0690 RCTUnsafeExecuteOnMainQueueSync + 52
29  KareOcrBusinessCards                0x0000000105557c8c -[RCTModuleData instance] + 816
30  KareOcrBusinessCards                0x00000001054f7254 __49-[RCTCxxBridge _prepareModulesWithDispatchGroup:]_block_invoke + 156
31  libdispatch.dylib                   0x000000010874c5a8 _dispatch_call_block_and_release + 32
32  libdispatch.dylib                   0x000000010874e05c _dispatch_client_callout + 20
33  libdispatch.dylib                   0x000000010875e914 _dispatch_main_queue_drain + 1456
34  libdispatch.dylib                   0x000000010875e354 _dispatch_main_queue_callback_4CF + 44
35  CoreFoundation                      0x00000001b86ce6f8 5CDC5D9A-E506-3740-B64E-BB30867B4F1B + 632568
36  CoreFoundation                      0x00000001b86b0058 5CDC5D9A-E506-3740-B64E-BB30867B4F1B + 507992
37  CoreFoundation                      0x00000001b86b4ed4 CFRunLoopRunSpecific + 612
38  GraphicsServices                    0x00000001f19b6368 GSEventRunModal + 164
39  UIKitCore                           0x00000001bab933d0 179501B6-0FC2-344A-B969-B4E3961EBE10 + 3810256
40  UIKitCore                           0x00000001bab93034 UIApplicationMain + 340
41  KareOcrBusinessCards                0x0000000104fc34ac main + 100
42  dyld                                0x00000001d6d20960 CB3FF411-4762-34D2-86A4-ECA13F9FB6C3 + 88416
2023-01-09 17:02:49.260049+0100 KareOcrBusinessCards[24531:3984910] [connection] nw_socket_handle_socket_event [C8:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:49.260208+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_connection_get_connected_socket_block_invoke [C8] Client called nw_connection_get_connected_socket on unconnected nw_connection
2023-01-09 17:02:49.260312+0100 KareOcrBusinessCards[24531:3984904] TCP Conn 0x281080320 Failed : error 0:61 [61]
2023-01-09 17:02:49.260652+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_connection_copy_connected_local_endpoint_block_invoke [C8] Client called nw_connection_copy_connected_local_endpoint on unconnected nw_connection
2023-01-09 17:02:49.260667+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_connection_copy_connected_remote_endpoint_block_invoke [C8] Client called nw_connection_copy_connected_remote_endpoint on unconnected nw_connection
2023-01-09 17:02:49.267323+0100 KareOcrBusinessCards[24531:3984931] [javascript] Running "KareOcrBusinessCards" with {"rootTag":1,"initialProps":{}}
2023-01-09 17:02:49.604161+0100 KareOcrBusinessCards[24531:3984695] [native] [GESTURE HANDLER] Initialize gesture handler for view <RCTRootContentView: 0x14de1b410; reactTag: 1; frame = (0 0; 390 844); gestureRecognizers = <NSArray: 0x282f18510>; layer = <CALayer: 0x2821aae40>>
2023-01-09 17:02:49.661374+0100 KareOcrBusinessCards[24531:3984695] [native] VisionCamera.didSetProps(_:): Updating 10 prop(s)...
2023-01-09 17:02:49.661510+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureCaptureSession(): Configuring Session...
2023-01-09 17:02:49.661553+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureCaptureSession(): Initializing Camera with device com.apple.avfoundation.avcapturedevice.built-in_video:6...
2023-01-09 17:02:49.661606+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureCaptureSession(): Adding Video input...
2023-01-09 17:02:49.671807+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureCaptureSession(): Adding Photo output...
2023-01-09 17:02:49.674678+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.invokeOnInitialized(): Camera initialized!
2023-01-09 17:02:49.674972+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureCaptureSession(): Session successfully configured!
2023-01-09 17:02:49.675826+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureFormat(): Configuring Format...
2023-01-09 17:02:49.675852+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureDevice(): Configuring Device...
2023-01-09 17:02:49.675962+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.configureDevice(): Device successfully configured!
2023-01-09 17:02:49.676173+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.didSetProps(_:): Starting Session...
2023-01-09 17:02:50.354931+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_socket_handle_socket_event [C10.1.1:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:50.355324+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_socket_handle_socket_event [C10.1.2:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:50.355722+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_connection_get_connected_socket_block_invoke [C10] Client called nw_connection_get_connected_socket on unconnected nw_connection
2023-01-09 17:02:50.355854+0100 KareOcrBusinessCards[24531:3984908] TCP Conn 0x281068140 Failed : error 0:61 [61]
2023-01-09 17:02:50.805392+0100 KareOcrBusinessCards[24531:3984910] [native] VisionCamera.didSetProps(_:): Started Session!
2023-01-09 17:02:52.368069+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_socket_handle_socket_event [C11.1.1:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:52.370401+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_socket_handle_socket_event [C11.1.2:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:52.371890+0100 KareOcrBusinessCards[24531:3984908] [connection] nw_connection_get_connected_socket_block_invoke [C11] Client called nw_connection_get_connected_socket on unconnected nw_connection
2023-01-09 17:02:52.372089+0100 KareOcrBusinessCards[24531:3984908] TCP Conn 0x281090000 Failed : error 0:61 [61]
2023-01-09 17:02:54.384086+0100 KareOcrBusinessCards[24531:3984905] [connection] nw_socket_handle_socket_event [C12.1.1:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:54.386362+0100 KareOcrBusinessCards[24531:3984905] [connection] nw_socket_handle_socket_event [C12.1.2:1] Socket SO_ERROR [61: Connection refused]
2023-01-09 17:02:54.387585+0100 KareOcrBusinessCards[24531:3984904] [connection] nw_connection_get_connected_socket_block_invoke [C12] Client called nw_connection_get_connected_socket on unconnected nw_connection
2023-01-09 17:02:54.387804+0100 KareOcrBusinessCards[24531:3984904] TCP Conn 0x2810883c0 Failed : error 0:61 [61]
flipper: Desktop failed to provide certificates. Error from flipper desktop:
rsocket::StreamInterruptedException: connection error
2023-01-09 17:02:57.559472+0100 KareOcrBusinessCards[24531:3984904] [native] VisionCamera.takePhoto(options:promise:): Capturing photo...
2023-01-09 17:02:57.561235+0100 KareOcrBusinessCards[24531:3984904] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[AVCapturePhotoOutput capturePhotoWithSettings:delegate:] settings.photoQualityPrioritization must not be higher than self.maxPhotoQualityPrioritization'
*** First throw call stack:
(0x1b863de88 0x1b196b8d8 0x1d1690df4 0x1058faabc 0x1058fcc0c 0x10874c5a8 0x10874e05c 0x10875610c 0x108756e34 0x108763cbc 0x205697df8 0x205697b98)
libc++abi: terminating with uncaught exception of type NSException
terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[AVCapturePhotoOutput capturePhotoWithSettings:delegate:] settings.photoQualityPrioritization must not be higher than self.maxPhotoQualityPrioritization'
(lldb)
```

### Device

iPhone 12 iOS 16.1.2

### VisionCamera Version

2.15.2

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by lylysunshinee** [comment]
IOS crash when setting qualityPrioritization = 'quality'. Any solution?

**Comment by myselfuser1** [comment]
Try this https://www.youtube.com/playlist?list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i

**Comment by mcaglarkeskin** [comment]
Just give `enableHighQualityPhotos={true}` to camera component. This will solve the problem.

**Comment by mrousavy** [comment]
Closing as this is a stale issue - this might have been fixed with the full rewrite in VisionCamera V3 (🥳) - if not, please create a new issue.

If your issue has been fixed, consider [sponsoring me on GitHub](https://github.com/sponsors/mrousavy) to say thanks 💖

**Comment by skizzo** [comment]
Unfortunately, XCode 15 is everything but stable when working with RN 0.71.x, so updating to v3 is not an option for me (and I guess many other devs)..

**Comment by PierreCapo** [comment]
FYI with vision-camera V3: 
The crash still occurs on iOS only if `qualityPrioritization: 'quality'` is set. As per mentioned by @mcaglarkeskin , setting `enableHighQualityPhotos={true}` fixes the issue.
